examples/src/main/java/com/mapbox/navigation/examples/core/MapMatchingActivity.kt

```diff
@@ -0,0 +1,396 @@
+package com.mapbox.navigation.examples.core
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.os.Looper
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.android.core.location.LocationEngine
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.android.core.location.LocationEngineResult
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.MapboxDirections
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.matching.v5.MapboxMapMatching
+import com.mapbox.api.matching.v5.models.MapMatchingResponse
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.location.LocationComponent
+import com.mapbox.mapboxsdk.location.LocationComponentActivationOptions
+import com.mapbox.mapboxsdk.location.modes.CameraMode
+import com.mapbox.mapboxsdk.location.modes.RenderMode
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.navigation.base.internal.accounts.UrlSkuTokenProvider
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.reroute.RerouteController
+import com.mapbox.navigation.core.reroute.RerouteState
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import com.mapbox.navigation.core.trip.session.TripSessionState.STARTED
+import com.mapbox.navigation.core.trip.session.TripSessionState.STOPPED
+import com.mapbox.navigation.core.trip.session.TripSessionStateObserver
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.utils.Utils
+import com.mapbox.navigation.examples.utils.Utils.PRIMARY_ROUTE_BUNDLE_KEY
+import com.mapbox.navigation.examples.utils.Utils.getRouteFromBundle
+import com.mapbox.navigation.examples.utils.extensions.toPoint
+import com.mapbox.navigation.ui.camera.DynamicCamera
+import com.mapbox.navigation.ui.camera.NavigationCamera
+import com.mapbox.navigation.ui.map.NavigationMapboxMap
+import com.mapbox.navigation.ui.map.NavigationMapboxMapInstanceState
+import com.mapbox.navigation.utils.internal.ifNonNull
+import java.lang.ref.WeakReference
+import kotlinx.android.synthetic.main.activity_basic_navigation_layout.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import timber.log.Timber
+
+/**
+ * This activity shows how to fetch DirectionsRoute with MapMatching
+ *
+ */
+@SuppressLint("MissingPermission")
+class MapMatchingActivity : AppCompatActivity(), OnMapReadyCallback {
+
+    private companion object {
+        private const val MAP_INSTANCE_STATE_KEY = "navigation_mapbox_map_instance_state"
```

Comment:
After https://github.com/mapbox/mapbox-navigation-android/pull/3436 this is handled differently and it's not needed anymore 👀 https://github.com/mapbox/mapbox-navigation-android/blob/master/examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavigationActivity.kt

Reply:
fixed